src/Tabs/TabsList.js

```diff
@@ -20,44 +16,52 @@ const TabsList = ({ children, xPadding = "none" }) => {
     tabsListRef,
     setIsResponsive,
     isResponsive,
+    setMarginRight,
   } = useContext(TabsContext);
   const childArray = React.Children.toArray(children);
 
-  const updateScrollButtonState = () => {
+  const updateTabPadding = () => {
     const { scrollWidth, clientWidth, scrollLeft } = tabsListRef.current;
-    const nextShowLeftArrow = scrollLeft > 1;
-    const nextShowRightArrow = scrollLeft < scrollWidth - clientWidth - 1;
-
-    setShowLeftArrow(nextShowLeftArrow);
-    setShowRightArrow(nextShowRightArrow);
-    setIsResponsive(nextShowLeftArrow || nextShowRightArrow);
+    setIsResponsive(
+      scrollLeft > 1 || scrollLeft < scrollWidth - clientWidth - 1
+    );
+    const tabsSize = getTabsSize();
```

Comment:
```suggestion
    const allFullyVisibleTabsWidth = getAllFullyVisibleTabsWidth();
```

While I like short names too. A more accurate name would allow for a lot less digging to understand the code.